pettingzoo/test/parallel_test.py

```diff
@@ -42,45 +42,57 @@ def parallel_api_test(par_env, num_cycles=1000):
         obs = par_env.reset()
         assert isinstance(obs, dict)
         assert set(obs.keys()) == (set(par_env.agents))
-        done = {agent: False for agent in par_env.agents}
+        terminated = {agent: False for agent in par_env.agents}
+        truncated = {agent: False for agent in par_env.agents}
         live_agents = set(par_env.agents[:])
         has_finished = set()
         for _ in range(num_cycles):
             actions = {
                 agent: sample_action(par_env, obs, agent)
                 for agent in par_env.agents
-                if agent in done and not done[agent]
+                if (
+                    (agent in terminated and not terminated[agent])
+                    or (agent in truncated and not truncated[agent])
+                )
             }
-            obs, rew, done, info = par_env.step(actions)
+            obs, rew, terminated, truncated, info = par_env.step(actions)
             for agent in par_env.agents:
-                assert agent not in has_finished, "agent cannot be revived once done"
+                assert agent not in has_finished, "agent cannot be revived once dead"
 
                 if agent not in live_agents:
                     live_agents.add(agent)
 
             assert isinstance(obs, dict)
             assert isinstance(rew, dict)
-            assert isinstance(done, dict)
+            assert isinstance(terminated, dict)
+            assert isinstance(truncated, dict)
             assert isinstance(info, dict)
 
             agents_set = set(live_agents)
-            keys = "observation reward done info".split()
-            vals = [obs, rew, done, info]
+            keys = "observation reward terminated truncated info".split()
+            vals = [obs, rew, terminated, truncated, info]
             for k, v in zip(keys, vals):
                 key_set = set(v.keys())
                 if key_set == agents_set:
                     continue
                 if len(key_set) < len(agents_set):
                     warnings.warn(f"Live agent was not given {k}")
                 else:
-                    warnings.warn(f"Agent was given {k} but was done last turn")
+                    warnings.warn(f"Agent was given {k} but was dead last turn")
 
             if hasattr(par_env, "possible_agents"):
                 assert set(par_env.agents).issubset(
                     set(par_env.possible_agents)
                 ), "possible_agents defined but does not contain all agents"
 
-                has_finished |= {agent for agent, d in done.items() if d}
+                has_finished |= {
+                    agent
+                    for agent, d in [
+                        (x[0], x[1] or y[1])
+                        for x, y in zip(terminated.items(), truncated.items())
+                    ]
+                    if d
+                }
```

Comment:
I'm just seeking some clarification here because I'm not completely familiar with the changes going on, so I'm ready to be educated. But are we certain that both terminated and truncated have the same number of items always?